app/src/main/kotlin/com/wire/android/ui/home/conversations/search/messages/SearchConversationMessagesResultsScreen.kt

```diff
@@ -18,29 +18,50 @@
 package com.wire.android.ui.home.conversations.search.messages
 
 import androidx.compose.foundation.lazy.LazyColumn
-import androidx.compose.foundation.lazy.items
 import androidx.compose.runtime.Composable
+import androidx.paging.compose.LazyPagingItems
+import androidx.paging.compose.itemContentType
+import androidx.paging.compose.itemKey
 import com.wire.android.ui.common.colorsScheme
+import com.wire.android.ui.home.conversations.AuthorHeaderHelper
 import com.wire.android.ui.home.conversations.MessageItem
 import com.wire.android.ui.home.conversations.info.ConversationDetailsData
-import com.wire.android.ui.home.conversations.mock.mockMessageWithText
 import com.wire.android.ui.home.conversations.model.UIMessage
-import com.wire.android.ui.theme.WireTheme
-import com.wire.android.util.ui.PreviewMultipleThemes
 
 @Composable
 fun SearchConversationMessagesResultsScreen(
-    searchResult: List<UIMessage>,
+    lazyPagingMessages: LazyPagingItems<UIMessage>,
     searchQuery: String = "",
     onMessageClick: (messageId: String) -> Unit
 ) {
     LazyColumn {
-        items(searchResult) { message ->
+        items(
+            count = lazyPagingMessages.itemCount,
+            key = lazyPagingMessages.itemKey { it.header.messageId },
+            contentType = lazyPagingMessages.itemContentType { it }
+        ) { index ->
+            val message: UIMessage = lazyPagingMessages[index]
+                ?: return@items // We can draw a placeholder here, as we fetch the next page of messages
+
+            val showAuthor = AuthorHeaderHelper.rememberShouldShowHeader(
+                index,
+                message,
+                lazyPagingMessages
+            )
+
+            val useSmallBottomPadding = AuthorHeaderHelper.rememberShouldHaveSmallBottomPadding(
+                index,
+                message,
+                lazyPagingMessages
+            )
```

Comment:
question: do you really want the searched messages to not have author and avatar if the message above is from the same author? I think on search screen each searched message should have author and not use small bottom padding. Also, this won't work if the order isn't reversed (`reverseLayout = true` for `LazyColumn`) because these functions from `AuthorHeaderHelper` need to check message above and below so if the order isn't reversed it will take wrong messages.

Reply:
I was not really sure if we should have kept it like this, but having a second look at the design it really doesn't make sense to have them grouped as for normal conversation list. Will be reverting this!